sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlGenerator.scala

```diff
@@ -16,164 +16,201 @@
  */
 package org.apache.spark.sql.catalyst.xml
 
+import java.io.Writer
 import java.sql.Timestamp
-import javax.xml.stream.XMLStreamWriter
+import javax.xml.stream.XMLOutputFactory
 
 import scala.collection.Map
 
+import com.sun.xml.txw2.output.IndentingXMLStreamWriter
```

Comment:
Hi, @sandip-db and @HyukjinKwon . Sorry for asking this question, but is this inevitable? This looks like the only instance importing `com.sun.*` in Apache Spark repository and which is added at Apache Spark 4.0.0-preview1.

```
$ git grep com.sun | grep import
sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlGenerator.scala:import com.sun.xml.txw2.output.IndentingXMLStreamWriter
```

Reply:
Yeah, I remember this is related to builtin package stuff in JDK library stuff right? Some JDKs don't have this IIRC (correct me if I am wrong). If we can, should be better to remove this usage.

Reply:
Thank you for answering, @HyukjinKwon . Let me file a JIRA in order not to forget this. I guess we need a careful look on this kind of part before Apache Spark 4.0.0 release.

Reply:
@sandip-db would you mind taking a look and followup when you find some time?

Reply:
Ack. I didn't find a drop-in replacement for IndentingXMLStreamWriter. I can look into implementing our own version.

Reply:
BTW, from JDK community side, this seems to be the unresolved issue.
- https://bugs.openjdk.org/browse/JDK-8141405 `Public API equivalent to com.sun.xml.txw2.output.IndentingXMLStreamWriter`

Reply:
I filed this. Please use this JIRA ID, @sandip-db .

- SPARK-48913 Avoid `com.sun.xml.txw2.output.IndentingXMLStreamWriter` usage